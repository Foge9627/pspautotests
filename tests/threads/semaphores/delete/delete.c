#include "../sub_shared.h"

SETUP_SCHED_TEST;

#define DELETE_TEST(title, sema) { \
	int result = sceKernelDeleteSema(sema); \
	if (result == 0) { \
		printf("%s: OK\n", title); \
	} else { \
		printf("%s: Failed (%X)\n", title, result); \
	} \
}

int main(int argc, char **argv) {
	// Verify scheduling order.
	SceUID sema = sceKernelCreateSema("delete1", 0, 0, 1, NULL);

	DELETE_TEST("Normal", sema);
	DELETE_TEST("NULL", 0);
	DELETE_TEST("Invalid", 0xDEADBEEF);
	DELETE_TEST("Deleted", sema);
	
	BASIC_SCHED_TEST(
		sceKernelDeleteSema(sema2);
	);

	return 0;
	
}