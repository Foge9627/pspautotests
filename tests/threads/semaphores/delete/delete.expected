Normal: OK
NULL: Failed (80020199)
Invalid: Failed (80020199)
Deleted: Failed (80020199)
ABCDE
